services/inc/str_util.h

```diff
@@ -87,4 +87,34 @@ inline bool endsWith(const char* str, const char* suffix) {
     return endsWith(str, strlen(str), suffix, strlen(suffix));
 }
 
+/**
+ * Escapes a set of characters by prepending them with an escape character.
+ *
+ * @param str Source string.
+ * @param strSize Size of the source string.
+ * @param spec Characters that need to be escaped.
+ * @param specSize Number of characters that need to be escaped.
+ * @param esc Escape character.
+ * @param dest Destination buffer.
+ * @param destSize Size of the destination buffer.
+ *
+ * @return Destination buffer.
+ */
+char* escape(const char* src, size_t srcSize, const char* spec, size_t specSize, char esc, char* dest, size_t destSize);
+
+/**
+ * Escapes a set of characters by prepending them with an escape character.
```

Comment:
Nice to see the method is well documented.  How does the caller know how many characters are added to the buffer. Is the output guaranteed to be null-terminated or only operates on the number of characters given?
 
For methods like this that are relatively simple, no side effects etc, I really liked doctest in python that allows you to write inline tests in documentation.  Seems like there is an equivalent for C++. https://github.com/onqtam/doctest